src/client/actions/ThreadListSync.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const Action = require('./Action');
+const Collection = require('../../util/Collection');
+const { Events } = require('../../util/Constants');
+
+class ThreadListSyncAction extends Action {
+  handle(data) {
+    const client = this.client;
+
+    const guild = client.guilds.cache.get(data.guild_id);
+    if (!guild) return {};
+
+    if (data.channels_ids) {
+      data.channel_ids.forEach(id => {
+        const channel = client.channels.resolve(id);
+        if (channel) this.removeStale(channel);
+      });
+    } else {
+      guild.channels.cache.forEach(channel => {
+        this.removeStale(channel);
+      });
+    }
+
+    const syncedThreads = new Collection();
+    data.threads.forEach(rawThread => {
+      const thread = client.channels.add(rawThread);
+      syncedThreads.set(thread.id, thread);
+    });
+
+    Object.values(data.members).forEach(rawMember => {
```

Comment:
You can either make a ternary with a for-of (`Array.isArray(data.members) ? data.members.entries() : Object.entries(data.members)`... or just a for-of loop